src/binder_call_forwarding.c

```diff
@@ -273,29 +273,56 @@ binder_call_forwarding_query_ok(
     const GBinderReader* args)
 {
     struct ofono_error err;
-    const RadioCallForwardInfo* infos;
     struct ofono_call_forwarding_condition* list = NULL;
     GBinderReader reader;
     gsize count = 0;
 
-    /* getCallForwardStatusResponse(RadioResponseInfo, vec<CallForwardInfo>) */
     gbinder_reader_copy(&reader, args);
-    infos = gbinder_reader_read_hidl_type_vec(&reader, RadioCallForwardInfo,
-        &count);
-    if (count) {
-        gsize i;
-
-        list = g_new0(struct ofono_call_forwarding_condition, count);
-        for (i = 0; i < count; i++) {
-            const RadioCallForwardInfo* info = infos + i;
-            struct ofono_call_forwarding_condition* fw = list + i;
-
-            fw->status = info->status;
-            fw->cls = info->serviceClass;
-            fw->time = info->timeSeconds;
-            fw->phone_number.type = info->toa;
-            memcpy(fw->phone_number.number, info->number.data.str,
-                MIN(OFONO_MAX_PHONE_NUMBER_LENGTH, info->number.len));
+
+    if (cbd->self->interface_aidl == RADIO_AIDL_INTERFACE_NONE) {
+        /* getCallForwardStatusResponse(RadioResponseInfo, vec<CallForwardInfo>) */
+        const RadioCallForwardInfo* infos = gbinder_reader_read_hidl_type_vec(
+            &reader, RadioCallForwardInfo, &count);
+        if (count) {
+            gsize i;
+
+            list = g_new0(struct ofono_call_forwarding_condition, count);
+            for (i = 0; i < count; i++) {
+                const RadioCallForwardInfo* info = infos + i;
+                struct ofono_call_forwarding_condition* fw = list + i;
+
+                fw->status = info->status;
+                fw->cls = info->serviceClass;
+                fw->time = info->timeSeconds;
+                fw->phone_number.type = info->toa;
+                memcpy(fw->phone_number.number, info->number.data.str,
+                    MIN(OFONO_MAX_PHONE_NUMBER_LENGTH, info->number.len));
+            }
+        }
+    } else {
+        /* getCallForwardStatusResponse(RadioResponseInfo, CallForwardInfo[] callForwardInfos) */
+        gbinder_reader_read_int32(&reader, &count);
```

Comment:
I assume with `int32` the returned value is `0...INT_MAX`? Or should the next check have `if (count > 0)` ?